api/python/t4/api.py

```diff
@@ -25,20 +25,34 @@ def copy(src, dest, meta=None):
 
 
 def put(obj, dest, meta=None):
-    if dest.endswith(AWS_SEPARATOR):
+    url = urlparse(dest)
+    if url.scheme != 's3':
+        raise NotImplementedError
+
+    bucket, path, version = parse_s3_url(url)
+    if version:
+        raise ValueError("Cannot push to a version")
+
+    if path.endswith(AWS_SEPARATOR):
```

Comment:
Minor thing, but it would be clearer to consistently use either '/' or AWS_SEPARATOR.